Sources/WalletConnectRelay/RelayClient.swift

```diff
@@ -291,4 +294,34 @@ public final class RelayClient {
             }
         }
     }
+
+    private func protectedSend(_ string: String) async throws {
+        return try await withCheckedThrowingContinuation { continuation in
+            protectedSend(string) { error in
+                if let error = error {
+                    continuation.resume(throwing: error)
+                } else {
+                    continuation.resume(returning: ())
+                }
+            }
+        }
+    }
+
+    private func protectedSend(_ string: String, timeout: Int = 5, completion: @escaping (Error?) -> Void) {
```

Comment:
I would move it to dispatcher.
Relay client is responsible for client<->relay interactions - subscribe/publish etc.
dispatcher is responsible for managing behaviour/policy of sending requests - unaware of the protocol methods etc.

and `protectedSend` to me sounds more like dispatcher responsibility.